### PR TITLE
Updated user-management.yaml to reflect the newly merged endpoints

### PR DESCRIPTION
**Pull request by dhelmert**
### Reason for this PR
- Many user endpoints for "students", "lecturers" and "admins" are now merged into users

### Changes in this PR
- Changed api to reflect upcoming changes in lagom api
- Fixed some inconsistent usage for trait objects

### CODE REVIEW

File: user-management.yaml [hunk 1]
```diff
@@ -87,110 +88,34 @@ paths:
...
+      summary: "Add a new user to the database"
+      description: "Adds a new user to the database after verification."
+      operationId: "addUser"
       security:
         - uc4_auth: [] 
       requestBody:
-        description: "Student object that needs to be added to the database"
+        description: "User object that needs to be added to the database"
         required: true
         content:
```
Comment (bastihav) [inline user-management.yaml:118]: can we add an example value here? the UI does not generate one by default

same for all other endpoints, that use the oneOf notation

File: user-management.yaml [hunk 2]
```diff
@@ -438,35 +308,28 @@ paths:
...
-        "422":
-          description: |
-            Unprocessable Entity  
-            types: validation error
-          content:
-            application/json:
-              schema:
-                $ref: '#/components/schemas/DetailedError'
+    
+  /role/{username}:
```
Comment (bastihav) [inline user-management.yaml]: can/should we now go back to 
`/users/{username}/role`
without ambiguous routes?

**Review by bastihav — APPROVED** [review]

File: user-management.yaml (hunk 1 above)
Comment (dhelmert) [reply user-management.yaml:118]: Example values are now added in POST, PUT, and GET on /users endpoint

File: user-management.yaml (hunk 2 above)
Comment (dhelmert) [reply user-management.yaml]: We can, and now we did. Thank you for the suggestion.

File: user-management.yaml [hunk 3]
```diff
@@ -1,7 +1,7 @@
 openapi: "3.0.0"
 info:
   description: "This is the User API for UC4."
-  version: "0.6.1"
+  version: "0.6.3"
```
Comment (bastihav) [inline user-management.yaml]:
```suggestion
  version: "0.7.1"
```

**Review by bastihav — APPROVED** [review]

**Pull request merged**